extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -278,9 +281,6 @@ protected DotName getDotName(BeanInfo bean) {
                 builder.addRemovalExclusion(new BeanClassAnnotationExclusion(annotation.getName()));
             }
         }
-        for (UnremovableBeanBuildItem exclusion : removalExclusions) {
-            builder.addRemovalExclusion(exclusion.getPredicate());
-        }
```

Comment:
So just to be extra sure, you handle that differently now?

Reply:
Yes, ``UnremovableBeanBuildItem``s are processed in the `io.quarkus.arc.deployment.ArcProcessor.validate()` phase and passed directly to the `BeanProcessor`. The advantage of this approach is that you can register an `UnremovableBeanBuildItem` based on e.g. an existing injection point.